packages/backend/src/database/migrations/20240409160248_add_dashboard_tab.ts

```diff
@@ -0,0 +1,129 @@
+import { file } from 'googleapis/build/src/apis/file';
+import { Knex } from 'knex';
+import { v4 as uuidv4 } from 'uuid';
+
+const tableName = 'dashboard_tabs';
+const dashboardTable = 'dashboards';
+const dashboardVersionTable = 'dashboard_versions';
+const DashboardTilesTableName = 'dashboard_tiles';
+const DashboardTileTabUuidColumnName = 'tab_uuid';
+
+type DbTabs = {
+    uuid: string;
+    name: string;
+    is_default: boolean;
+    dashboard_id: number;
+    dashboard_version_id: number;
+};
+
+export async function up(knex: Knex): Promise<void> {
+    // create dashboard tab table
+    if (!(await knex.schema.hasTable(tableName))) {
+        await knex.schema.createTable(tableName, (tableBuilder) => {
+            tableBuilder.string('name').notNullable();
+            tableBuilder.uuid('uuid').notNullable().defaultTo(uuidv4());
+            tableBuilder.boolean('is_default');
+            tableBuilder
+                .integer('dashboard_id')
+                .notNullable()
+                .references('dashboard_id')
+                .inTable(dashboardTable)
+                .onDelete('CASCADE');
+            tableBuilder
+                .integer('dashboard_version_id')
+                .notNullable()
+                .references('dashboard_version_id')
+                .inTable(dashboardVersionTable)
+                .onDelete('CASCADE');
+            tableBuilder
+                .timestamp('created_at', { useTz: false })
+                .notNullable()
+                .defaultTo(knex.fn.now());
+            tableBuilder
+                .timestamp('updated_at', { useTz: false })
+                .notNullable()
+                .defaultTo(knex.fn.now());
+            tableBuilder.unique([
+                'uuid',
+                'dashboard_id',
+                'dashboard_version_id',
+            ]);
+        });
+
+        // add tab_uuid column which reference to the dashboard tab
+        if (
+            !(await knex.schema.hasColumn(
+                DashboardTilesTableName,
+                DashboardTileTabUuidColumnName,
+            ))
+        ) {
+            await knex.schema.alterTable(
+                DashboardTilesTableName,
+                (tableBuilder) => {
+                    tableBuilder
+                        .uuid(DashboardTileTabUuidColumnName)
+                        .nullable();
+                },
+            );
+        }
+
+        // create a default tab for each dashboard
```

Comment:
We don't need to create a default tab. 

If dashboard has no tabs -> we render like we do now
If there are tabs -> we default tiles with no tab uuid to the first tab ( we need to track tab order somewhere )
If user deletes the last tab -> we render like we do now